### PR TITLE
Add SDK Tracking

### PR DESCRIPTION
**Pull request by masojus**
Adds the "Keen-Sdk" HTTP header.
- Also, centralize assembly attributes.
- Add a script to update the assembly versions and nuspec version as well as build and pack.

### CODE REVIEW

File: Keen.NET.Test/Keen.NET.Test.csproj [hunk 1]
```diff
@@ -83,6 +83,12 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="..\SharedAssemblyInfo.cs">
+      <Link>Properties\SharedAssemblyInfo.cs</Link>
```
Comment (masojus) [inline Keen.NET.Test/Keen.NET.Test.csproj]: They are there: https://github.com/keenlabs/keen-sdk-net/pull/43/files#diff-de185a19d58a705bb8adced8c3dab3d8

They are in the root of the solution and linked in for each project, so you won't actually see them in the ...\Properties\ folders on disk.

Comment (josephwegner) [inline Keen.NET.Test/Keen.NET.Test.csproj]: derp. Sorry

File: Keen.NET_35/Event.cs [hunk 2]
```diff
@@ -25,6 +25,7 @@ public JArray GetSchemas()
                 var client = new RestClient(_serverUrl);
                 var request = new RestRequest("", Method.GET);
                 request.AddHeader("Authorization", _prjSettings.MasterKey);
+                request.AddHeader("Keen-Sdk", KeenUtil.GetSdkVersion());
```
Comment (baumatron) [inline Keen.NET_35/Event.cs]: Maybe we should define a constant for "Keen-Sdk".

File: NewSemVer.ps1 [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+function Get-MajMinPatchVersion([string] $version) {
+    return $version.Split('-'.ToCharArray(), 2)[0]
+}
+ 
+<#
+Effect version updates in SharedVersionInfo.cs.
+#>
+function Update-AssemblyVersionAttributes ([string] $version) {
+    $majMinPatchPattern = '[0-9]+(\.([0-9]+|\*)){1,3}'
+	$assemblyVersionPattern = "AssemblyVersion\(`"$majMinPatchPattern`"\)"
```
Comment (baumatron) [inline NewSemVer.ps1]: Tabs vs. spaces are inconsistent in this file.

File: Keen.NET_35/Event.cs (hunk 2 above)
Comment (masojus) [inline Keen.NET_35/Event.cs]: I was planning to put "Keen-Sdk", "Authorization", "content-type", "application/json" and so forth in constants in whatever file where we implement an HttpClientFactory in an upcoming change.

File: NewSemVer.ps1 (hunk 3 above)
Comment (masojus) [inline NewSemVer.ps1]: Good call. Probably a copy/paste error. Fixed.

**Pull request merged**